store/postgres/src/copy.rs

```diff
@@ -744,6 +754,24 @@ impl Connection {
                 if table.is_cancelled(&self.conn)? {
                     return Ok(Status::Cancelled);
                 }
+
+                // Pause copying if replication is lagging behind to avoid
+                // overloading replicas
+                let mut lag = catalog::replication_lag(&self.conn)?;
+                if lag > MAX_REPLICATION_LAG {
```

Comment:
This is equivalent to a `while lag > MAX_REPLICATION_LAG` loop.

Reply:
No, it's not, since we only get out of the loop when lag falls below `ACCEPTABLE_REPLICATION_LAG`; getting in and out of the loop is asymmetric, we basically hope that if lag is between `ACCEPTABLE_REPLICATION_LAG` and `MAX_REPLICATION_LAG` that it will resolve itself.

Reply:
> This is equivalent to a `while lag > MAX_REPLICATION_LAG` loop.

It's not, since we only get out of the loop when lag falls below ACCEPTABLE_REPLICATION_LAG; getting in and out of the loop is asymmetric, we basically hope that if lag is between ACCEPTABLE_REPLICATION_LAG and MAX_REPLICATION_LAG that it will resolve itself.

Reply:
Oh, right! I didn't catch that. Marking as resolved :smile: